src/constants.jl

```diff
@@ -0,0 +1,236 @@
+const StringList = Ptr{Cstring}
+
+@enum(
+    GDALDataType,
+    GDT_Unknown = 0,
+    GDT_Byte = 1,
+    GDT_UInt16 = 2,
+    GDT_Int16 = 3,
+    GDT_UInt32 = 4,
+    GDT_Int32 = 5,
+    GDT_Float32 = 6,
+    GDT_Float64 = 7,
+    GDT_CInt16 = 8,
+    GDT_CInt32 = 9,
+    GDT_CFloat32 = 10,
+    GDT_CFloat64 = 11,
+    GDT_TypeCount = 12,
+)
+
+@enum(
+    OGRFieldType,
+    OFTInteger = 0,
+    OFTIntegerList = 1,
+    OFTReal = 2,
+    OFTRealList = 3,
+    OFTString = 4,
+    OFTStringList = 5,
+    OFTWideString = 6,
+    OFTWideStringList = 7,
+    OFTBinary = 8,
+    OFTDate = 9,
+    OFTTime = 10,
+    OFTDateTime = 11,
+    OFTInteger64 = 12,
+    OFTInteger64List = 13,
+    OFTMaxType = 14, # 13
+)
+
+@enum(
+    OGRFieldSubType,
+    OFSTNone = 0,
+    OFSTBoolean = 1,
+    OFSTInt16 = 2,
+    OFSTFloat32 = 3,
+    OFSTJSON = 4,
+    OFSTMaxSubType = 5, # 4
+)
```

Comment:
> How is this file currently made?

This file is manually created/maintained, mostly from me looking up the corresponding `src/common.jl` file in GDAL.jl.

>  CEnum allows multiple fields to have the same value.

I think that's desired behavior for GDAL.jl, but awkward when it came to type-checking enums in this package (I forgot the specific example I had in mind, sorry; but it feels weird that enums aren't mutually exclusive). And yeah, I think the values for `wkb*` can overshoot the bounds of valid values for enums.

For all those reasons, I figured it's better to maintain an independent set of enums that

(i) [behaves like enums](https://github.com/yeesian/ArchGDAL.jl/blob/5107fe5fa14b92496eeca2cc58736c7f49cd2653/src/constants.jl), and
(ii) [can be converted to GDAL enums](https://github.com/yeesian/ArchGDAL.jl/blob/5107fe5fa14b92496eeca2cc58736c7f49cd2653/src/types.jl#L228-L534) when they're being passed to GDAL.jl (based on this [convert macro](https://github.com/yeesian/ArchGDAL.jl/blob/5107fe5fa14b92496eeca2cc58736c7f49cd2653/src/utils.jl#L1-L46) which replaces the old gdalenum macro)

Reply:
Ok I understand how this is working now, thanks. What do you mean with "behaves like enums", do CEnums not behave like enums? I think I would make the choice to stick with the original values to avoid confusion, though I understand your reasoning here as well.

Reply:
> What do you mean with "behaves like enums"

Oh, I was thinking mostly about having distinct values. I'm starting to remember that it might have to do with e.g. using them as keys in a dictionary and accidentally overriding the entries, resulting in unintuitive behavior:

```julia
julia> Dict(GDAL.GCI_YCbCr_CrBand => "a", GDAL.GCI_Max => "b")
Dict{GDAL.GDALColorInterp, String} with 1 entry:
  GCI_YCbCr_CrBand => "b"
```

Between warning people to use `string(...)` for CEnums when using them as keys versus having enums that enforce distinct values, I opted for the latter.

Reply:
> I think I would make the choice to stick with the original values to avoid confusion, though I understand your reasoning here as well.

Happy to write it up in the developer documentation if it might become a source of confusion/contention -- since the value of `GDAL.GCI_Max` might change from one version to the next, my reasoning was that we'd want to be using the "names" of enums rather than knowing about (much less using) the corresponding values that are used to model them as enums.

There are some cases ([[1]](https://github.com/yeesian/ArchGDAL.jl/blob/5107fe5fa14b92496eeca2cc58736c7f49cd2653/src/types.jl#L551-L559), [[2]](https://github.com/yeesian/ArchGDAL.jl/blob/5107fe5fa14b92496eeca2cc58736c7f49cd2653/src/types.jl#L536-L541)) where the underlying values are of interest, so the CEnum approach in GDAL.jl is still very useful to keep. But since that's already provided by GDAL.jl, I figured it'll be complementary/helpful for end-users to have a package which models the corresponding Enum (rather than CEnum) behavior and facilitate bi-directional conversion.

Reply:
Great, sounds good. Instead of dev docs I would also be happy with a short comment above all the enums that their value is different from GDAL but they map correctly with at-convert.

Reply:
Documented this discussion in https://github.com/yeesian/ArchGDAL.jl/pull/179/commits/9c7069cbd30b4bf61fbe59901577a18c1950301f.

Reply:
> Instead of dev docs I would also be happy with a short comment above all the enums that their value is different from GDAL but they map correctly with at-convert.

Added in https://github.com/yeesian/ArchGDAL.jl/pull/179/commits/d5a5885057a53499965c83839c6aba5912e67191. Leaving this comment thread open for you to resolve or let me know if I missed anything